java/src/main/java/ai/rapids/cudf/BinaryOperable.java

```diff
@@ -76,6 +76,25 @@ static DType implicitConversion(BinaryOperable lhs, BinaryOperable rhs) {
     if (a == DType.BOOL8 || b == DType.BOOL8) {
       return DType.BOOL8;
     }
+    if (a.isDecimalType() && b.isDecimalType()) {
+      // Here scale is created with value 0 as `scale` is required to create DType of
+      // decimal type. Dtype is discarded for binary operations for decimal types in cudf as a new
+      // DType is created for output type with new scale. New scale for output depends upon operator.
+      int scale = 0;
```

Comment:
Please let me know if this is okay. I have added comment why we are setting the scale to `0`.

Reply:
Sorry, I think the comment wasn't clear enough. I don't think we need to determine the scale of result here. 
From other discussion below regarding output type which is ignored, please let me know what would be the preferred way to handle this?

Reply:
Please update the javadocs for this function as well, and then I think it is OK.